File: datalib/src/main/java/org/vihv/contoureditor/Contour.java
package org.vihv.contoureditor;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

@Entity
@Table(name="contour")
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class Contour {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    public Long id = null;
    public String title;

    @OneToMany(cascade = CascadeType.ALL)
    public List<Vertex> vertexes = new ArrayList<>();
}

File: web/src/main/ts/model/Contour.ts
import { Vertex } from "./Vertex";

export class Contour {
    public title:string="";
    public x:number = 0;
    public y:number = 0;
    public vertexes: Vertex[] = [];

    public constructor(title:string, vertexes:Vertex[]) {
        this.title = title;
        this.vertexes = vertexes;
    }
}